supabase_functions/_async/functions_client.py

```diff
@@ -30,15 +30,33 @@ def __init__(
             "User-Agent": f"supabase-py/functions-py v{__version__}",
             **headers,
         }
-        self._client = AsyncClient(
-            base_url=self.url,
-            headers=self.headers,
-            verify=bool(verify),
-            timeout=int(abs(timeout)),
-            proxy=proxy,
-            follow_redirects=True,
-            http2=True,
-        )
+
+        if timeout is not None:
+            warn("The 'timeout' parameter is deprecated. Please configure it in the httpx client instead.", DeprecationWarning, stacklevel=2)
+        if verify is not None:
+            warn("The 'verify' parameter is deprecated. Please configure it in the httpx client instead.", DeprecationWarning, stacklevel=2)
+        if proxy is not None:
+            warn("The 'proxy' parameter is deprecated. Please configure it in the httpx client instead.", DeprecationWarning, stacklevel=2)
+
+        self.verify = bool(verify) if verify is not None else True
+        self.timeout = int(abs(timeout)) if timeout is not None else 60
+
+        if httpx is not None:
+            self._client = httpx
+            self._client.base_url = self.url
+            self._client.headers = self.headers
+            self._client.follow_redirects = True
```

Comment:
I don't believe these params should be set besides the `base_url` and `headers`. The others should be handled by the developer when they decide to use their own `httpx` client.